src/mlpack/core/data/bow_encoding_impl.hpp

```diff
@@ -0,0 +1,138 @@
+/**
+ * @file bow_impl.hpp
+ * @author Jeffin Sam
+ *
+ * Implementation of Bag of Words encoding functions.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_BOW_ENCODING_IMPL_HPP
+#define MLPACK_CORE_DATA_BOW_ENCODING_IMPL_HPP
+
+// In case it hasn't been included yet.
+#include "bow_encoding.hpp"
+
+namespace mlpack {
+namespace data {
+
+void Bow::Reset()
+{
+  mappings.clear();
+  originalStrings.clear();
+}
+
+Bow::Bow(const Bow& oldObject)
+{
+  originalStrings = oldObject.originalStrings;
```

Comment:
I think it's better to move this line to the initializer list.

Reply:
Not sure, what does this mean ?

Reply:
I mean
```
Bow::Bow(const Bow& oldObject) :
    originalStrings(oldObject.originalStrings)
{
}
```

Reply:
Thanks :), Done.